daemon/job/snapjob.go

```diff
@@ -0,0 +1,379 @@
+	a.tasksMtx.Lock()
+	defer a.tasksMtx.Unlock()
+	var copy snap_activeSideTasks
+	copy = a.tasks
+	if u == nil {
+		return copy
+	}
+	u(&copy)
+	a.tasks = copy
+	return copy
+}
+
+
+type snap_activeMode interface {
+	Type() Type
+	RunPeriodic(ctx context.Context, wakeUpCommon chan<- struct{})
+	//FSFilter() endpoint.FSFilter
+	FSFilter() zfs.DatasetFilter
+}
+
+
+
+type modeSnap struct {
+//    fsfilter         endpoint.FSFilter
+    fsfilter         zfs.DatasetFilter
+    snapper *snapper.PeriodicOrManual
+}
+
+func (m *modeSnap) Type() Type { return TypeSnap }
+
+func (m *modeSnap) RunPeriodic(ctx context.Context, wakeUpCommon chan <- struct{}) {
+    m.snapper.Run(ctx, wakeUpCommon)
+}
+
+func (m *modeSnap) FSFilter() zfs.DatasetFilter {
+	return m.fsfilter
+}
+
+func modeSnapFromConfig(g *config.Global, in *config.SnapJob) (*modeSnap, error) {
+    m := &modeSnap{}
+    fsf, err := filters.DatasetMapFilterFromConfig(in.Filesystems)
+    if err != nil {
+        return nil, errors.Wrap(err, "cannnot build filesystem filter")
+    }
+    m.fsfilter = fsf
+
+    if m.snapper, err = snapper.FromConfig(g, fsf, in.Snapshotting); err != nil {
+        return nil, errors.Wrap(err, "cannot build snapper")
+    }
+
+    return m, nil
+}
+
+
+func snap_activeSide(g *config.Global, in *config.SnapJob, mode *modeSnap) (j *snap_ActiveSide, err error) {
+
+	j = &snap_ActiveSide{mode: mode}
+	j.name = in.Name
+/***	j.promRepStateSecs = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace:   "zrepl",
+		Subsystem:   "replication",
+		Name:        "state_time",
+		Help:        "seconds spent during replication",
+		ConstLabels: prometheus.Labels{"zrepl_job":j.name},
+	}, []string{"state"})
+	j.promBytesReplicated = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   "zrepl",
+		Subsystem:   "replication",
+		Name:        "bytes_replicated",
+		Help:        "number of bytes replicated from sender to receiver per filesystem",
+		ConstLabels: prometheus.Labels{"zrepl_job":j.name},
+	}, []string{"filesystem"})
+
+	j.clientFactory, err = connecter.FromConfig(g, in.Connect)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot build client")
+	}
+***/
+	j.promPruneSecs = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace:   "zrepl",
+		Subsystem:   "pruning",
+		Name:        "time",
+		Help:        "seconds spent in pruner",
+		ConstLabels: prometheus.Labels{"zrepl_job":j.name},
+	}, []string{"prune_side"})
+	j.prunerFactory, err = pruner.NewSinglePrunerFactory(in.Pruning, j.promPruneSecs)
+	//j.rules = pruning.RulesFromConfig(in.Keep)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot build snapjob pruning rules")
+	}
+	return j, nil
+}
+
+func (j *snap_ActiveSide) RegisterMetrics(registerer prometheus.Registerer) {
+	registerer.MustRegister(j.promPruneSecs)
+}
+
+/*func (j *ActiveSide) Name() string { return j.name }
+
+type ActiveSideStatus struct {
```

Comment:
please define a new status struct `SnapJobStatus` for SnapJob, which contains the `pruner.Report`.
We can omit reports on the snapshotting for now.